### PR TITLE
Con 2747 prevent co e journey

### PR DESCRIPTION
**Pull request by Howesy77**

### CODE REVIEW

File: src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Mappers/Apprentice/DetailsViewModelMapper.cs [hunk 1]
```diff
@@ -82,11 +85,13 @@ public async Task<DetailsViewModel> Map(DetailsRequest source)
                     HasPendingChangeOfPartyRequest = pendingChangeOfPartyRequest != null,
                     PendingChangeOfPartyRequestWithParty = pendingChangeOfPartyRequest?.WithParty,
                     HasApprovedChangeOfPartyRequest = approvedChangeOfPartyRequest != null,
+                    HasPendingChangeOfProviderRequest = pendingChangeOfProviderRequest != null,
                     EncodedNewApprenticeshipId = approvedChangeOfPartyRequest?.NewApprenticeshipId != null
                         ? _encodingService.Encode(approvedChangeOfPartyRequest.NewApprenticeshipId.Value,
                             EncodingType.ApprenticeshipId)
                         : null,
                     IsContinuation = data.Apprenticeship.IsContinuation && data.Apprenticeship.PreviousProviderId == source.ProviderId,
+                    HasContinuation = data.Apprenticeship.HasContinuation,
```
Comment (hkf-tech) [inline src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Mappers/Apprentice/DetailsViewModelMapper.cs:94]: Is it possible to use the Approved flag (Change of party request) like on task CON-2585 to hide the link if it is a previous provider and there is a completed CoP reqeuest. Then we don't need the das-commitment changes.

File: src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Mappers/Apprentice/DetailsViewModelMapper.cs [hunk 2]
```diff
@@ -82,11 +85,13 @@ public async Task<DetailsViewModel> Map(DetailsRequest source)
                     HasPendingChangeOfPartyRequest = pendingChangeOfPartyRequest != null,
                     PendingChangeOfPartyRequestWithParty = pendingChangeOfPartyRequest?.WithParty,
                     HasApprovedChangeOfPartyRequest = approvedChangeOfPartyRequest != null,
+                    HasPendingChangeOfProviderRequest = pendingChangeOfProviderRequest != null,
```
Comment (hkf-tech) [inline src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Mappers/Apprentice/DetailsViewModelMapper.cs:88]: can we please add a unit test for this

File: src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.IntegrationTests/SFA.DAS.ProviderCommitments.IntegrationTests.csproj [hunk 3]
```diff
@@ -16,7 +16,7 @@
     <PackageReference Include="nunit" Version="3.11.0" />
     <PackageReference Include="NUnit3TestAdapter" Version="3.11.0" />
     <PackageReference Include="Microsoft.NET.Test.Sdk" Version="15.9.0" />
-    <PackageReference Include="SFA.DAS.CommitmentsV2.Api.Client" Version="4.4.0" />
+    <PackageReference Include="SFA.DAS.CommitmentsV2.Api.Client" Version="4.14.0" />
```
Comment (hkf-tech) [inline src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.IntegrationTests/SFA.DAS.ProviderCommitments.IntegrationTests.csproj:19]: may not be required

**Review by hkf-tech — APPROVED** [review]

File: src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Mappers/Apprentice/DetailsViewModelMapper.cs (hunk 2 above)
Comment (Howesy77) [reply src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Mappers/Apprentice/DetailsViewModelMapper.cs:88]: Done

File: src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Mappers/Apprentice/DetailsViewModelMapper.cs (hunk 1 above)
Comment (hkf-tech) [inline src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Mappers/Apprentice/DetailsViewModelMapper.cs:94]: please add unit test for this.

Comment (hkf-tech) [inline src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Mappers/Apprentice/DetailsViewModelMapper.cs:94]: please add unit test for this.

**Review by hkf-tech — CHANGES_REQUESTED** [review]

File: src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Mappers/Apprentice/DetailsViewModelMapper.cs (hunk 1 above)
Comment (Howesy77) [reply src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Mappers/Apprentice/DetailsViewModelMapper.cs:94]: Done

**Review by hkf-tech — APPROVED** [review]

**Pull request merged**